File: php/binary-search/BinarySearch.php
<?php

class BinarySearch
{
    /**
     * 二分查找元素在数组中的位置。
     *
     * @param int $value 待查找的元素。
     * @param array $array 待查找的数组。
     *
     * @return int -1 表示没有找到元素，否则为元素在数组中的索引。
     */
    public static function search(int $value, array $array): int
    {
        $low = 0;
        $high = count($array) - 1;

        while ($low <= $high) {
            $middle = floor(($high + $low) / 2);

            if ($array[$middle] === $value) {
                return $middle;
            }

            if ($array[$middle] > $value) {
                $high = $middle - 1;
            } else {
                $low = $middle + 1;
            }
        }

        return -1;
    }
}

File: java/binary-search/src/main/java/binarysearch/BinarySearch.java
package binarysearch;

public class BinarySearch {
    public static int search(int value, int[] array) {
        int low = 0;
        int high = array.length - 1;

        while (low <= high) {
            int middle = (low + high) / 2;

            if (array[middle] == value) {
                return middle;
            } else if (array[middle] > value) {
                high = middle - 1;
            } else {
                low = middle + 1;
            }
        }

        return -1;
    }
}
